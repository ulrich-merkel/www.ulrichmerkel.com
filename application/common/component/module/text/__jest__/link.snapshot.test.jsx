/* eslint-disable import/no-extraneous-dependencies, func-names */
import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import ModuleTextLink from '../link';

describe('common/component/module/text/link', function () {
    const defaultProps = {
        content: {
            linkTo: '/foo',
            linkLabel: 'Module text link label',
            linkTitle: 'Module text link title'
        }
    };

    it('should render correctly', function () {
        const tree = renderer.create(
            <MemoryRouter>
                <ModuleTextLink
                    {...defaultProps}
                />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should return null if text is empty', function () {
        const tree = renderer.create(
            <MemoryRouter>
                <ModuleTextLink
                    {...defaultProps}
                    content={null}
                />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
