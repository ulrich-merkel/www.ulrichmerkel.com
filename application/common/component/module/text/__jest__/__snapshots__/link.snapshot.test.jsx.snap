// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/component/module/text/link should render correctly 1`] = `
<div
  className="m-text__link is-centered"
>
  <a
    aria-current={null}
    className="c-link"
    href="/foo"
    onClick={[Function]}
    tabIndex="0"
    title="Module text link title"
  >
    <i
      aria-hidden="true"
      className="c-font-icon--earth"
    />
    Module text link label
  </a>
</div>
`;

exports[`common/component/module/text/link should return null if text is empty 1`] = `null`;
